doc/Dataplane Code Gen/DataplaneCodeGen_Quickstart_Tutorial.md

```diff
@@ -0,0 +1,185 @@
+- Remove the `Azure.Template.DataplaneCodeGen.Tests` project from the solution.
+
+**Note**: `Azure.<your-sdk-name>` is the Azure SDK package name.
+
+**Learn more:** see the [.NET Guidelines on Namespace Naming](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-namespace-naming) for more information on naming Azure SDK packages. This name diverges from approved Azure SDK namespaces, since it is intended to be used just for this tutorial.
+
+#### 4. Save and commit your changes to your branch
+
+This will make it easier to see the changes you make when you generate code in the clean solution.
+
+```
+% git add -A
+% git commit -m "initial commit for <service-name>"
+```
+
+### Create the generated layer
+
+In this section, we'll create a generated API layer built on Azure Core. We'll use a small swagger file containing a REST API definition for Azure Messaging WebPubSub.
+
+**Learn more:** the [autorest.csharp README](https://github.com/Azure/autorest.csharp#setup) has great samples showing how to customize the API in the generated code. Explore this further as you design APIs for your own service.
+
+#### 1. Set the Swagger File
+
+Open autorest.md in your `Azure.<your-sdk-name>` project. In the input-file definition, replace the file path with your swagger link, for example: [WebPubSub swagger link](https://github.com/Azure/azure-rest-api-specs/blob/39c7d63c21b9a29efe3907d9b949d1c77b021907/specification/webpubsub/data-plane/WebPubSub/stable/2021-10-01/webpubsub.json). Update a namespace definition to set the namespace C# generator will use when creating your types. Update the AAD token credential(for AAD token, refer [this](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/template-LLC/Azure.Template.LLC/src/autorest.md#autorest-configuration) example) or Azure key credential (for Azure Key credential, refer [this](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/webpubsub/Azure.Messaging.WebPubSub/src/autorest.md#swagger-sources) example). Save your changes.
+
+Your `autorest.md` file should now look like:
+
+````md
+# `Azure.<your-sdk-name>` Code Generation
+
+Run `dotnet build /t:GenerateCode` to generate code.
+
+### AutoRest Configuration
+> see https://aka.ms/autorest
+
+``` yaml
+input-file:
+- <swagger-file-link>
+namespace: <namespace>
+public-clients: true
+low-level-client: true
+security: AADToken
+security-scopes: <AAD-token-scope>
+```
+````
+
+**Learn more:** see the [.NET Guidelines for Namespaces](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-namespace-naming) to learn more about namespaces in the Azure SDK.
+
+#### 2. Run the Generate Command
+
+From a PowerShell command prompt, navigate to the directory holding `Azure.<your-sdk-name>.sln`. Run the following commands:
+
+```
+sdk\<your-service-name>\Azure.<your-sdk-name>\src> dotnet msbuild /t:GenerateCode
+```
+
+#### 3. Explore the Generated Code
```

Comment:
I'm wondering if we can restructure this a bit to move the Explore part forward before the reader does the generation.  If we do, we can walk them through what's in the project before we have them start deleting/renaming things and refer to areas we know will be there.

Reply:
I kept this section here but also, I added the description at the top(second paragraph) of this section about the autorest.md and generated folder [here](https://github.com/Azure/azure-sdk-for-net/blob/9981a1f3c18dca95f186fe2adb8283e3c5caeb10/doc/Data%20Plane%20Code%20Generation/AzureSDKCodeGeneration_DataPlane_Quickstart.md#create-the-generated-layer).